packages/components/src/List/ListComposition/README.md

```diff
@@ -0,0 +1,362 @@
+* with new needs, adding a div inside can be complicated
+* with different needs, having different layouts is impossible
+* we pass all the props for all the subcomponents making the apropscalypse we have today
+
+The result is a component that is really complicated to use and maintain, and not flexible at all.
+
+*Example*: how would you develop a filter bar that display current filters between the toolbar and the VirtualizedList ?
+I guess you would create a div, insert it if we have a flag prop, or configuration props, and the filters to display will be added to the current apropcalypse, making the api even more heavier.
+
+## Principle
+
+Let's try to make it flexible. What do we need ?
+* a component where we can compose subcomponents
+* possibility to adapt the layout for different needs
+* all subcomponents must work with the List, and have synchronized data with other related subcomponents. For example, the sort order is synchronized between the VirtualizedList and Sort toolbar element
+* simplify the components api
+
+Nicolas Maligne started a PoC to turn the List into Compound components.
+
+### Simple example: Tabs
+
+```javascript
+<Tabs
+    items={[{ id: 0, label: 'First link' }, { id: 1, label: 'Second link' }]}
+    selectedId={0}
+    onChange={callback}
+/>
+```
+
+```javascript
+<Tabs onChange={callback}>
+    <Tab id={0} selected>First link</Tab>
+    <Tab id={1}>Second link</Tab>
+</Tabs>
+```
+
+### Complex example: The List
+
+Before
+```javascript
+const toolbarProps = {
+    actionBar: {
+        actions: {
+            left: [
+                { id: 'add', label: 'Add Folder' },
+                { id: 'remove', label: 'Remove Folder' },
+            ],
+        },
+    },
+    display: {
+        onChange: this.onDisplayChange,
+    },
+    sort: {
+        field: 'name',
+        onChange: action('sort.onChange'),
+        options: [{ id: 'id', name: 'Id' }, { id: 'name', name: 'Name' }],
+    }
+};
+const listProps = {
+    columns: [
+        { key: 'id', label: 'Id' },
+        { key: 'name', label: 'Name' },
+        { key: 'author', label: 'Author' },
+        { key: 'created', label: 'Created' },
+        { key: 'modified', label: 'Modified' },
+    ],
+    items: [{ id: 0 }, { id: 1 }, { id: 2 }, { id: 3 }],
+    titleProps: {
+        key: 'name',
+        iconKey: 'icon',
+        displayModeKey: 'display',
+        onClick: action('onTitleClick'),
+        onEditCancel: action('onEditCancel'),
+        onEditSubmit: action('onEditSubmit'),
+    },
+    itemProps: {
+        classNameKey: 'className',
+    },
+};
+
+<List
+    id="talend"
+	displayMode="table"
+	list={listProps}
+	toolbar={toolbarProps}
+/>
+```
+
+After
+```javascript
+const toolBar = (
+    <List.Toolbar>
+        <ActionBar left={[
+            { id: 'add', label: 'Add Folder' },
+            { id: 'remove', label: 'Remove Folder' },
+        ]} />
+        <List.DisplayMode id="my-list-displayMode" />
```

Comment:
Don't we want to enforce the relation of the compound components ?
`
<List>
<List.Toolbar>
<Toolbar.DisplayMode />
</List.Toolbar>
</List>
`

Reply:
IMHO it's better, it enforces visually how the composition works.
And can emphasize the css relation wihout opening code.

Reply:
I prefer to stay with 1 unique prefix. Just reading your example, I wonder where Toolbar group comes from ? Does it work properly with the List ? Does it use the same context for synchro ? etc